### PR TITLE
Feature/hamza docs

### PR DESCRIPTION
**Pull request by htahir1**
## Describe changes
First draft of docs changes. There are one or two pages in progress but reviewers can already see whats up

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: docs/book/user-guide/advanced-guide/data-versioning-and-lineage.md [hunk 1]
```diff
@@ -1,25 +1,73 @@
...
+Each time a ZenML pipeline runs, the system first checks if there have been any changes in the inputs, outputs, parameters, or configuration of the pipeline steps. Each step in a run gets a new directory:
 
-Under the hood, all the artifacts in our ML pipeline are automatically stored in an Artifact Store. By default, this is simply a place in your local file system, but we could also configure ZenML to store this data in a cloud bucket like Amazon S3 or any other place instead. We will see this in more detail when we migrate our MLOps stack to the cloud in a later chapter.
+![Visualizing artifacts](../../.gitbook/assets/zenml_artifact_store_underthehood_1.png)
 
-## Orchestrators
+If a step is new or has been modified, ZenML creates a new directory structure in the [Artifact Store](../component-galery/artifact-stores/README.md) with a unique ID and stores the data using the appropriate materializers in this directory.
```
Comment (AlexejPenner) [inline docs/book/user-guide/advanced-guide/data-versioning-and-lineage.md:15]: ... new directory structure ... - i feel this is too detailed

Comment (htahir1) [reply docs/book/user-guide/advanced-guide/data-versioning-and-lineage.md:15]: Really? i think its useful

Comment (AlexejPenner) [reply docs/book/user-guide/advanced-guide/data-versioning-and-lineage.md:15]: looking at it now I am also not too sure what my issue with this was. Maybe the wording? 

If a step is new or has been modified, ZenML creates a new **folder** in the ...

Maybe using the word folder it sounds better already

Comment (htahir1) [reply docs/book/user-guide/advanced-guide/data-versioning-and-lineage.md:24]: Ok ill see

**Pull request merged**